src/ngraph_encapsulate_op.cc

```diff
@@ -292,6 +292,59 @@ class NGraphEncapsulateOp : public OpKernel {
 
       auto function_size = ng_function->get_graph_size() / 1024;  // kb unit
 
+      // Get the serialized ops and stored the allreduce ops to a vector and
+      vector<shared_ptr<ng::Node>> allreduce_op_list;
```

Comment:
It would be better to use nGraph class `NodeVector`. https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/node_vector.hpp

Reply:
Changed